TLM/TLM/Geometry/Impl/SegmentGeometry.cs

```diff
@@ -898,30 +898,13 @@ public class SegmentGeometry : IEquatable<SegmentGeometry> {
 			return contains;
 		}
 
-        /// <summary>
-        /// Determines if, according to the stored geometry data, the segment is eligible for turn-on-red.
-        /// 
-        /// A segment geometry verification is not performed.
-        /// </summary>
-        /// <returns>true, if eligible for turn-on-red</returns>
-        public bool HasValidTurnOnRedOutgoingSegment(SegmentEndGeometry segmentEndGeometry) {
-            if (Constants.ServiceFactory.SimulationService.LeftHandDrive) {
-                return (segmentEndGeometry.NumOutgoingLeftSegments > 0 ||
-                    (IsOneWay() && (segmentEndGeometry.NumOutgoingLeftSegments > 0 || segmentEndGeometry.NumOutgoingRightSegments > 0 && segmentEndGeometry.OutgoingOneWay)));
-            } else {
-                return (segmentEndGeometry.NumOutgoingRightSegments > 0 ||
-                    (IsOneWay() && (segmentEndGeometry.NumOutgoingLeftSegments > 0 && segmentEndGeometry.OutgoingOneWay  || segmentEndGeometry.NumOutgoingRightSegments > 0)));
-            }
-        }
-
-
```

Comment:
*Geometry classes should not contain complex "business" logic. They should only describe geometry related things.